docroot/sites/oniowa.uiowa.edu/modules/oniowa_core/oniowa_core.module

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for Oniowa Core module.
+ */
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_form_alter().
+ */
+function oniowa_core_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Use the form['#id'] over $form_id because it is more specific.
+  switch ($form['#id']) {
+    case 'views-exposed-form-events-page-upcoming':
+      $cid = 'sitenow_events:event:dates';
+      $data = \Drupal::cache()->get($cid);
+      if (!$data) {
+        $options = [];
+        $options[''] = t('Select a date');
+        $query = \Drupal::entityQuery('node');
+        $query->condition('type', 'event')
+          ->condition('status', 1)
+          ->condition('field_event_when.end_value', time(), '>')
+          ->sort('field_event_when', 'ASC');
+        $result = $query->execute();
+        if ($result) {
+          $nodes = \Drupal::entityTypeManager()
+            ->getStorage('node')
+            ->loadMultiple($result);
+          foreach ($nodes as $node) {
+            $date = $node->field_event_when->value;
+            if ($date) {
+              $day = date('l, F j', $date);
+              if (!isset($options[$date])) {
+                $options[$date] = $day;
+              }
+            }
+          }
+        }
+        // Cache the options, but set them to be invalidated
+        // should any events be CUDed.
```

Comment:
Its not immediately clear what `CUD` might stand for, so we may want to spell this out.

Reply:
CreatedUpdatedDeleted?

Reply:
```suggestion
        // should any events be Created/updated/deleted.
```